ml-agents/mlagents/trainers/trainer_util.py

```diff
@@ -10,9 +10,50 @@
 from mlagents.trainers.bc.offline_trainer import OfflineBCTrainer
 
 
-def initialize_trainers(
-    trainer_config: Dict[str, Any],
-    external_brains: Dict[str, BrainParameters],
+class TrainerFactory:
```

Comment:
Not a big deal, but this could just be a function which returns the generator function (would involve less repetition)--
```
def create_trainer_factory(
    trainer_config: Any,
    summaries_dir: str,
    run_id: str,
    model_path: str,
    keep_checkpoints: int,
    train_model: bool,
    load_model: bool,
    seed: int,
    meta_curriculum: MetaCurriculum = None,
    multi_gpu: bool = False,
):
    def generate(brain_parameters: BrainParameters) -> Trainer:
        ...
    return generate
```

Reply:
I think I will leave it like it is for now. Is this really problematic ?

Reply:
Not problematic, just a little harder to change later.  It's fine to leave it